### PR TITLE
Exp pdcli

### PR DESCRIPTION
**Pull request by Krishnanunni333**
https://issues.redhat.com/browse/OSD-8114

### CODE REVIEW

**Comment by chopraapooja** [comment]
@Krishnanunni333  Could you please mention the Jira ticket number in the description.

File: LICENSE [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
                                  Apache License
```
Comment (chopraapooja) [inline LICENSE:2]: don't need this

File: cmd/root.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
+/*
+Copyright © 2021 NAME HERE <EMAIL ADDRESS>
```
Comment (chopraapooja) [inline cmd/root.go:2]: Mention Red Hat, Inc.

**Comment by Krishnanunni333** [comment]
applied the changes

**Comment by chopraapooja** [comment]
@Krishnanunni333 don't delete the LICENCE file... don't make any changes to LICENCE file

File: cmd/root.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var cfgFile string
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "pdcli",
+	Short: "A brief description of your application",
```
Comment (chopraapooja) [inline cmd/root.go:30]: Please fill in description

File: cmd/root.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+}
+
+func init() {
+	cobra.OnInitialize(initConfig)
+
+	// Here you will define your flags and configuration settings.
+	// Cobra supports persistent flags, which, if defined here,
+	// will be global for your application.
+
+	rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.pdcli.yaml)")
```
Comment (chopraapooja) [inline cmd/root.go:55]: We don't need config & toggle flag

File: cmd/root.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+	// Cobra also supports local flags, which will only run
+	// when this action is called directly.
+	rootCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
+}
+
+// initConfig reads in config file and ENV variables if set.
+func initConfig() {
+	if cfgFile != "" {
```
Comment (chopraapooja) [inline cmd/root.go:64]: Remove this for now. We don't need to init viper config, remove viper from the package for now

**Comment by chopraapooja** [comment]
@Krishnanunni333 could you please make the directory structure similar to https://github.com/openshift-online/ocm-cli
With updated README.md
And makefile

@supreeth7 @MitaliBhalla Could you please help in here to improve App scaffolding

**Comment by chopraapooja** [comment]
CC: @dofinn

File: LICENSE (hunk 1 above)
Comment (chopraapooja) [reply LICENSE:2]: don't need any changes in License file

File: cmd/root.go [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	cobra.CheckErr(rootCmd.Execute())
+}
+
+func init() {
+	cobra.OnInitialize(initConfig)
+
+	// Here you will define your flags and configuration settings.
```
Comment (chopraapooja) [inline cmd/root.go:53]: could you please add `root.AddCommand(login.Cmd)` here

File: cmd/root.go [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"github.com/spf13/cobra"
+)
+
+var cfgFile string
```
Comment (chopraapooja) [inline cmd/root.go:22]: don't need this var

File: cmd/root.go [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "pdcli",
+	Short: "A CLI application called pdcli short for pagerDuty CLI.",
+	Long: `A CLI application called pdcli short for pagerDuty CLI. It can be used reduce the time taken, from the time, SRE receives a PD alert to the time where troubleshooting on the cluster actually begins. 
+	
+	There are 3 main sub commands:
+
```
Comment (chopraapooja) [inline cmd/root.go:31]: you don't need to mention sub commands over here, it will be auto populated by cobra

File: cmd/root.go [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+	// Here you will define your flags and configuration settings.
+	// Cobra supports persistent flags, which, if defined here,
+	// will be global for your application.
+
+	// Cobra also supports local flags, which will only run
+	// when this action is called directly.
+}
+
+// initConfig reads in config file and ENV variables if set.
+func initConfig() {
```
Comment (chopraapooja) [inline cmd/root.go:62]: remove initConfig

**Comment by chopraapooja** [comment]
Could you provide an example test in test file

File: cmd/pdcli/login/cmd.go [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+// pdcli/login/cmdCmd represents the pdcli/login/cmd command
+var pdcli/login/cmdCmd = &cobra.Command{
+	Use:   "pdcli/login/cmd",
```
Comment (chopraapooja) [inline cmd/pdcli/login/cmd.go:26]: this should be `login`

File: cmd/pdcli/login/cmd.go [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+Cobra is a CLI library for Go that empowers applications.
+This application is a tool to generate the needed files
+to quickly create a Cobra application.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println("pdcli/login/cmd called")
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(pdcli/login/cmdCmd)
```
Comment (chopraapooja) [inline cmd/pdcli/login/cmd.go:40]: AddCommand will be called in root.go

**Pull request merged**